From d4b46cdd3cf0b7e9eaa415fc87a4281695f2349c Mon Sep 17 00:00:00 2001
From: Lujin Wang <luwang@nvidia.com>
Date: Mon, 12 Aug 2019 12:03:29 -0700
Subject: [PATCH] libweston: Remove source repaint_timer in
 weston_compositor_shutdown

Fixes a definitely lost:
== 56 bytes in 1 blocks are definitely lost in loss record 16 of 45
==    at 0x48450F8: malloc (vg_replace_malloc.c:309)
==    by 0x4B55E93: wl_event_loop_add_timer (event-loop.c:197)
==    by 0x4126CF: weston_compositor_create (in /usr/local/bin/weston)
==    by 0x409997: main (in /usr/local/bin/weston)

Signed-off-by: Lujin Wang <luwang@nvidia.com>
Signed-off-by: Pekka Paalanen <pekka.paalanen@collabora.com>
---
 libweston/compositor.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libweston/compositor.c b/libweston/compositor.c
index c2da3a48c..90321e860 100644
--- a/libweston/compositor.c
+++ b/libweston/compositor.c
@@ -7733,6 +7733,7 @@ weston_compositor_shutdown(struct weston_compositor *ec)
 	struct weston_output *output, *next;
 
 	wl_event_source_remove(ec->idle_source);
+	wl_event_source_remove(ec->repaint_timer);
 
 	/* Destroy all outputs associated with this compositor */
 	wl_list_for_each_safe(output, next, &ec->output_list, link)
-- 
GitLab

